.github/workflows/test-external.yml

```
@@ -55,19 +54,22 @@ jobs:
           # Use the virtual environment created by the build step (name comes from build config)
           source ci-test/bin/activate
           uv pip list
-          nohup llama stack run tests/external-provider/llama-stack-provider-ollama/run.yaml --image-type ${{ matrix.image-type }} > server.log 2>&1 &
+          nohup llama stack run tests/external/run-byoa.yaml --image-type ${{ matrix.image-type }} > server.log 2>&1 &
 
       - name: Wait for Llama Stack server to be ready
         run: |
+          echo "Waiting for Llama Stack server..."
           for i in {1..30}; do
-            if ! grep -q "Successfully loaded external provider remote::custom_ollama" server.log; then
-              echo "Waiting for Llama Stack server to load the provider..."
-              sleep 1
-            else
-              echo "Provider loaded"
+            if curl -s http://localhost:8321/v1/health | grep -q "OK"; then
+              echo "Llama Stack server is up!"
               exit 0
             fi
+            sleep 1
           done
-          echo "Provider failed to load"
+          echo "Llama Stack server failed to start"
           cat server.log
           exit 1
+
+      - name: Test external API
+        run: |
+          curl -s http://localhost:8321/v1/weather/locations
```

We can improve this `curl` test and surface errors in a better way:

```suggestion
          curl -sSf http://localhost:8321/v1/weather/locations
```

`-S`: When  used with `-s`, curl shows an error message if it fails.
`-f`: Exit `curl` with a non-zero exit code when the server returns HTTP response codes 400 or greater.